### PR TITLE
refactor(config) : access definition of customs from basis

### LINKED ISSUE #1738: Make visible definition (args) of custom(logger|reporter) to it's base class and decorators
When i start to implement #1737 - i have found that where is not simple way to make such behavior at BaseLauncher or launcherDecorator level - because they are instanced and used without access to `args` of customLauncher (reporter) through injector, while it was `local` for custom factory.

While custom factory is not requied to do something with `args` they are used just in .ctor method on customer way and then is lost (so it is in karma-firefox-launcher for example).

So while #1737 is based on idea that 'binpath' will be accessible from BaseLauncher - it must be way to supply source definition (from config) to it.
It's not only about #1737 - it can be different scenarios when we could want to modify/setup core launcher/reporter behavior from config for any customLauncher(reporter) of any base type.

Have implement it in my fork and will prepare update soon. I have not found way to do it with injector and made it strait by defining instance variable with special name at factory level and supply null-safe accessor to it at BaseLauncher(Reporter) level.

### PR DESCRIPTION
**Pull request by comdiv**
Allows custom(reporter|launcher)'a args to be accessible from
Launcher(Reporter)Base

Closes #1738

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 6c8e2a23c24e7cbd1aec7efa947bf58afb031226
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at http://karma-runner.github.io/0.13/dev/git-commit-msg.html

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 3588081f855c1169d03aceafcd5850571bfe5236
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at http://karma-runner.github.io/0.13/dev/git-commit-msg.html

---
This message was auto-generated by https://gitcop.com

File: lib/config.js [hunk 1]
```diff
@@ -195,7 +195,14 @@ var normalizeConfig = function (config, configFilePath) {
...
 
       module[type + ':' + name] = ['factory', function (injector) {
-        return injector.createChild([locals], [token]).get(token)
+        var result = injector.createChild([locals], [token]).get(token)
+        // we cannot force external launchers to keep args somewhere
+        // and use it in creation, but we require to keep args for base
+        // classes/decorators usage
+        // while it stores in locals - it will be not simple DI way
+        // to access when launcher will be created, so setup it explicitly
+        result.$definition = result.$definition || definition
```
Comment (Dignifiedquire) [inline lib/config.js]: Not a big fan of the `$`. What exactly do you want to symbolise with it?

**Comment by Dignifiedquire** [comment]
One nit pick, otherwise looks good

**Comment by dignifiedquire** [comment]
@comdiv ping, this needs a rebase onto the latest master.

**Pull request closed without merging**